flang/include/flang/ISO_Fortran_binding.h

```diff
@@ -160,8 +164,15 @@ typedef struct CFI_cdesc_t {
 #endif
 
 #ifdef __cplusplus
-  RT_API_ATTRS inline bool HasAddendum() const { return extra & 1; }
-  RT_API_ATTRS inline int GetAllocIdx() const { return ((int)extra >> 1); }
+  RT_API_ATTRS inline bool HasAddendum() const {
+    return extra & _CFI_ADDENDUM_FLAG;
+  }
+  RT_API_ATTRS inline void SetHasAddendum() {
+    extra = extra | _CFI_ADDENDUM_FLAG;
+  }
+  RT_API_ATTRS inline int GetAllocIdx() const {
+    return ((int)extra >> _CFI_ADDENDUM_BITS);
```

Comment:
This will have to change later if another field is added.  Might also want to have a member function that sets the field.

Reply:
Added.